pkg/models/batch_transformation.go

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2020-present Snowplow Analytics Ltd.
+ * All rights reserved.
+ *
+ * This software is made available by Snowplow Analytics, Ltd.,
+ * under the terms of the Snowplow Limited Use License Agreement, Version 1.0
+ * located at https://docs.snowplow.io/limited-use-license-1.0
+ * BY INSTALLING, DOWNLOADING, ACCESSING, USING OR DISTRIBUTING ANY PORTION
+ * OF THE SOFTWARE, YOU AGREE TO THE TERMS OF SUCH LICENSE AGREEMENT.
+ */
+
+package models
+
+import "time"
+
+// MessageBatch houses batches of messages, for batch transformations to operate across
+type MessageBatch struct {
+	OriginalMessages    []*Message        // Most targets will use the data from here, but where we have a http templating transformation, we would use this to ack batches of messages
+	BatchData           []byte            // Where we template http requests, we use this to define the body of the request
+	HTTPHeaders         map[string]string // For dynamic headers feature
```

Comment:
I am not a _huge_ fan of having a `target` specific implementation mixed in with a generic model here. Is there another way to carry this data through without it being specifically mapped to `http`?

Would something like `Parameters map[string]interface{}` where in `http` you can then access a `HTTPHeaders` key within the `Parameters` work to abstract this away?

Reply:
I don't love it either. This actually comes from having implemented the dynamic header transformation - which has [the same problem for the message model](https://github.com/snowplow/snowbridge/blob/c8c94f8a13367260f9a1a280cdd5d86098f66521/pkg/models/message.go#L23).

I considered that way of doing things but it's a trade-off - the downside of that approach is that you have a more obscure API and the target's logic depends on that specific key, but the api defines it as being anything. 

I experimented with other things we could do but didn't find an elegant solution (yet), and it didn't feel like it serves the project well to labour on it for too long. 

Right now we only have one thing that needs to do this, so my thinking was that this will do for the moment but when we need to design for further similar things we should revisit the api design.

I'm not massively opposed to doing what you suggest either - I just haven't given up on finding something better

Reply:
Fair point! As long as the seed of "maybe we should change this" is planted I am fine with it staying where it is.

Reply:
Not even that much of a maybe to be honest! Just needs to percolate a bit. Perhaps we will even see the answer when the rest of this refactor falls into place.